app/org/sagebionetworks/bridge/services/ParticipantService.java

```diff
@@ -397,8 +397,6 @@ public void resendVerification(Study study, ChannelType type, String userId) {
         } else if (type == ChannelType.PHONE && participant.getPhone() != null) {
             AccountId accountId = AccountId.forPhone(study.getIdentifier(), participant.getPhone());
             accountWorkflowService.resendVerificationToken(type, accountId);
-        } else {
-            throw new UnsupportedOperationException("Channel type not implemented");
```

Comment:
What's the problem with throwing this exception? If we're not passing in a channel, this is almost certainly a bug in the server. If this happens, it would be better to fast-fail the request with a 500 Internal Server Error than to do nothing and mask the error.

Reply:
You get a 500 error back if the participant does not have an email or phone address, which just leaks information about the account. This isn't a pure test of the enumeration handling alone.

Reply:
Okay, I see it now. It's the compound clauses on the if statements. I think you need nested if statements.